marin/export/hf_upload.py

```diff
@@ -0,0 +1,259 @@
+import dataclasses
+import functools
+import logging
+import os
+import tempfile
+import time
+from dataclasses import dataclass
+from functools import wraps
+
+import fsspec
+import humanfriendly
+from fsspec.implementations.local import LocalFileSystem
+from huggingface_hub import create_commit, upload_folder
+from tqdm_loggable.auto import tqdm
+
+from marin.execution import ExecutorStep, InputName
+from marin.utils import fsspec_glob
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class UploadToHfConfig:
+    input_path: str | InputName
+    repo_id: str
+    repo_type: str = "dataset"
+    token: str | None = None
+    revision: str | None = None
+
+    upload_kwargs: dict[str, str] = dataclasses.field(default_factory=dict)
+    """Will be passed to huggingface_hub.upload_folder"""
+    private: bool = False
+    commit_batch_size: str = "1GiB"
+    small_file_limit: str = "5 MiB"
+    """
+    The size limit for small files. Files larger than this will be uploaded using lfs.
+    """
+
+
+def upload_dir_to_hf(
+    input_path: str | InputName,
+    repo_id: str,
+    repo_type: str = "dataset",
+    token: str | None = None,
+    certificate_path: str | None = None,
+    private: bool = False,
+    revision: str | None = None,
+    commit_batch_size: str = "1GiB",
+    **upload_kwargs: str,
+) -> ExecutorStep:
+    """
+    Uploads a path (possibly a GCS path) to a Hugging Face repo.
```

Comment:
Does that mean that @Helw150 should use this script for uploading checkpoints without downloading first?

Reply:
Happy to integrate this with the model checkpoint stuff when you feel ready! I made the model checkpoints an executor step in https://github.com/marin-community/marin/pull/1244 but it still DL's to the workers temp dir.